Tau HLT study

skeshri opened the pull request:
Hi @kandrosov and @ballmond,

I have made a PR for updated hlt config file for Tau HLT study. Kindly have a look.

Regards,
Sumit

kandrosov commented:
This version is still not working, right? Would you be available to prepare a separate PR with only YAML configurations for MC and data datasets that need to be processed?

skeshri commented:
Hi @kandrosov and @ballmond,

I have added the latest MC samples and data. Please have a look.
The only samples that I don't find are:
`
VBFHHto2B2Tau_CV-1_C2V-1_C3-1: /VBFHHto2B2Tau_CV-1_C2V-1_C3-1_TuneCP5_13p6TeV_madgraph-pythia8/Run3Winter23Digi-126X_mcRun3_2023_forPU65_v1-v2/GEN-SIM-RAW
VBFHHto2B2Tau_CV-1_C2V-1_C3-2: /VBFHHto2B2Tau_CV-1_C2V-1_C3-2_TuneCP5_13p6TeV_madgraph-pythia8/Run3Winter23Digi-126X_mcRun3_2023_forPU65_v1-v2/GEN-SIM-RAW
VBFHHto2B2Tau_CV-1_C2V-2_C3-1: /VBFHHto2B2Tau_CV-1_C2V-2_C3-1_TuneCP5_13p6TeV_madgraph-pythia8/Run3Winter23Digi-126X_mcRun3_2023_forPU65_v1-v2/GEN-SIM-RAW
`
Otherwise, I have updated all the corresponding samples.
Please let me know if you need anything else.

kandrosov commented:
Thank you @skeshri. Could you please perare a separate PR with only Run3Winter24_HLT yamls? or undo changes for other files in this PR.

brallmond commented:
Hi @kandrosov 
I reviewed Sumit's changes. All MC/Data for Run3Summer23BPix and Run3Winter24 are in their respective yaml files (minus some unavailable VBF To HH files) and the changes to other files have been reverted.

kandrosov commented on Production/plugins/PixelTrackTableProducer.cc:
```diff
@@ -13,7 +13,8 @@
 #include "DataFormats/TrajectoryState/interface/LocalTrajectoryParameters.h"
```
revert changes

skeshri replied:
If we revert "#include "RecoPixelVertexing/PixelTrackFitting/interface/FitUtils.h" then the compilation will fail in CMSSW_13_3_X. Do you want to revert it?

kandrosov commented on Production/crab/Run3Summer23BPixDRPremix_HLT/mc.yaml:
```diff
@@ -0,0 +1,29 @@
+
+
+
+DYTo2L_MLL-4to50: /DYTo2L_MLL-4to50_TuneCP5_13p6TeV_pythia8/Run3Summer23BPixDRPremix-130X_mcRun3_2023_realistic_postBPix_v2-v1/GEN-SIM-RAW 
```
Do we want both Run3Summer23BPix and Run3Winter24?

skeshri replied:
I have added samples from campaign "Run3Summer23BPixDRPremix" after @brallmond requested.
@brallmond, can you please answer. Thanks!

kandrosov reviewed (changes requested):
thank you @skeshri and @brallmond! Here are some remaining comments/questions to address.

skeshri commented:
@kandrosov and @brallmond , should I remove zerobias datasets and include only era D?

brallmond commented:
For the comments above, @sumit, what you should do is `git checkout` the original changed files (everything but the yamls containing the changes) instead of manually reverting them, that way there won't be commits in the repository history that effectively do nothing. 

For the Data samples, we don't need EphemeralZeroBias because we aren't doing L1 studies, like Konstantin said. For EphemeralHLTPhysics, lets use Era D only since it is the most recent. Sorry I neglected to notice only part of C was used previously. 

For the MC samples, I think we should run both Summer23BPix and Winter24. 
The reason is Summer23BPix is as close to the start of 2024 conditions as possible, and Winter24 is conditions projected midway through data taking. It would be ideal to have a set of WPs for both so that we can work ahead of time instead of doing another sample campaign in a month from now.

skeshri commented:
@kandrosov and @brallmond, I'm closing this PR to have a cleaner one as Braden suggested. I'm making a fresh PR now.
